users/stdiscosrv.rst

```diff
@@ -205,20 +205,46 @@ allows:
 - Use of a subdomain name without requiring a port number added to the URL
 - Sharing an SSL certificate with multiple services on the same server
 
+Note that after this configuration, clients have to omit the :code`?id=...`
+parameter of the discovery server URL on their configuration. Client-side
+validation will be done by checking the visible proxy server's certificate.
+
 Requirements
 ^^^^^^^^^^^^
 
 - Run the discovery server using the -http flag  :code:`stdiscosrv -http`.
 - SSL certificate/key configured for the reverse proxy
-- The "X-Forwarded-For" http header must be passed through with the client's
+- The "X-Forwarded-For" HTTP header must be passed through with the client's
   real IP address
-- The "X-SSL-Cert" must be passed through with the PEM-encoded client SSL
-  certificate
+- The "X-SSL-Cert" HTTP header must be passed through with the PEM-encoded
+  client SSL certificate
 - The proxy must request the client SSL certificate but not require it to be
   signed by a trusted CA.
 
+Apache
+""""""
+Add the following lines to the configuration:
+
+.. code-block:: apache
+
+    SSLProxyEngine On
+    SSLVerifyClient optional_no_ca
+    # The following is required, otherwise syncthing-discosrv
+    # outputs 'no certificates' when a POST is issued on it
+    RequestHeader set X-SSL-Cert "%{SSL_CLIENT_CERT}s"
+    # Apparently the RemoteIPHeader directive is not really required.
+    # I couldn't understand why it is recommended for a simple reverse proxy
+    # with no other services in front.
+    # Using this would require using a2enmod remoteip
+    #RemoteIPHeader X-Forwarded-For
```

Comment:
This is required, because that's how discovery server gets remote peers address, which it then uses when telling others where to find the device.

Reply:
So, I was studying this a bit more by sniffing what Apache is sending back do stdiscosrv.

On the POST requests, I see the following headers:
```
    127.0.0.1.56506 > 127.0.0.1.8443: (...)
(...)
........POST / HTTP/1.1                                                                                                                                                   
Host: localhost:8443                           
User-Agent: Go-http-client/1.1                                                                
Content-Type: application/json                                                                                                                                               
Accept-Encoding: gzip                                                                                                                                                               
X-SSL-Cert: -----BEGIN CERTIFICATE----- (...) -----END CERTIFICATE-----                                                                                              
X-Forwarded-For: <visible IP of the client>                                                                                                                                         
X-Forwarded-Host: <public address of my Apache server>                                                           
X-Forwarded-Server: <public address of my Apache server>                                                                                                                                           
Connection: Keep-Alive                                                                         
Content-Length: 118                                                                                                                                                            
                                                                                              
{"addresses":["relay://<public address of my Apache server>:22067/?id=("]}                                                              
```

On GET requests, I see the following:
```
    127.0.0.1.56414 > 127.0.0.1.8443: (...)

.qF..qF.GET /?device=(...) HTTP/1.1
Host: localhost:8443                                                  
User-Agent: Go-http-client/1.1   
Accept-Encoding: gzip                  
X-SSL-Cert: (null)                                
X-Forwarded-For: <visible IP of the client>
X-Forwarded-Host: <public address of my Apache server>                                             
X-Forwarded-Server: <public address of my Apache server>                                                   
Connection: Keep-Alive                                                               
```

So, this leads me to conclude that apparently httpd adds the `X-Forwarded-For` field by default, without any extra configuration.
Should we still recommend a more complex configuration (running a2enmod + adding that line) given that, at least on httpd 2.4.38, this is not needed by default?

I'm more inclined to add this explicit information there, so that users may want to try with no a2enmod+RemoteIPHeader, and if they fail, they can try adding it.

Reply:
I have no idea, you're the apache specialist here.

Reply:
Heh, fair enough. I updated the PR with this information, hopefully making it simple enough that you won't be annoyed by too many support requests regarding it.